### PR TITLE
feat(Extend config passing configWrapper): Wrap config with passed configWrapper

### PR DESCRIPTION
**Pull request by iamgarima**

### CODE REVIEW

File: server/routes.js [hunk 1]
```diff
@@ -131,13 +131,15 @@ function getDomainSlug(publisherConfig, hostName) {
...
-  publisherConfig = require("./publisher-config")
+  publisherConfig = require("./publisher-config"),
+  configWrapper = config => config
 ) {
   return function withConfig(f, staticParams) {
     return function (req, res, next) {
       const client = getClient(req.hostname);
       return client
         .getConfig()
+        .then(config => configWrapper(config))
```
Comment (sharangj) [inline server/routes.js:142]: Shouldnt this be a await or a promise?

Comment (iamgarima) [reply server/routes.js:142]: Yes, it returns a Promise which is handled by the next `then`

Comment (sharangj) [reply server/routes.js:142]: I meant shouldnt it be like this. `await config wrapper(config)`

Comment (iamgarima) [reply server/routes.js:142]: I don't think so. Nested promises return the final resolved value.

Run the following code:

 // Checking nested Promise behaviour
 const nestedPromise = () => {
   return new Promise((resolve, reject) => {
     return resolve(new Promise((resolve, reject) => resolve(5)));
   });
 }

 const value = nestedPromise().then(result => {
   console.log("Result =>", result);
   return result;
  });

Comment (sivteck) [reply server/routes.js:142]: if the function passed to the then function returns a `Promise`, as it happens here when the `configWrapper` function is called, the next `then` in the promise chain waits till it is resolved, in this case `await` inside the resolve function is not necessary

ref: https://javascript.info/promise-chaining#summary

Comment (sharangj) [reply server/routes.js:142]: But this is true only if the function returns a promise. Say we dont return a promise?

File: test/integration/config-test.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      Promise.resolve({
+        "publisher-settings": {
+          title: "Madrid"
+        }
+      }),
+  };
+};
+
+const configWrapperStub = config => {
+  return {
```
Comment (sharangj) [inline test/integration/config-test.js:22]: Can you create a test with an async function?

File: server/routes.js (hunk 1 above)
Comment (iamgarima) [reply server/routes.js:142]: We don't need to do any of it. Usually this is how we implement it. Please let me know how u want me to update the current implementation. I will make the changes

File: test/integration/config-test.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      .get("/route-data.json")
+      .then((res) => {
+        const response = JSON.parse(res.text);
+        assert.equal("bar", response.data.config.extendedConfig.foo);
+      })
+      .then(done);
+  });
+});
+
+describe("configWrapper", () => {
```
Comment (sharangj) [inline test/integration/config-test.js:66]: Added 2 different describes. Please add it in the same describe

**Review by sharangj — APPROVED** [review]

**Pull request merged**